libpod/kube.go

```diff
@@ -0,0 +1,276 @@
+package libpod
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containers/libpod/pkg/lookup"
+	"github.com/containers/libpod/pkg/rootless"
+	"github.com/containers/libpod/pkg/util"
+	"github.com/cri-o/ocicni/pkg/ocicni"
+	"github.com/ghodss/yaml"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	KubeNotSupportedError = errors.New("This libpod function is not supported in kube translation yet")
+)
+
+// InpectforKube takes a slice of libpod containers and generates
+// one v1.Pod description that includes just a single container.
+func InspectForKube(containers []*Container) error {
+	var output []byte
+	developmentComment := []byte("# Generation of Kubenetes YAML is still under development!\n")
+	logrus.Warn("This function is still under heavy development.")
+	// Iterate each container
+	for _, ctr := range containers {
+		// Generate the v1.Pod yaml description
+		pod, err := simplePodWithV1Container(ctr)
+		if err != nil {
+			return err
+		}
+		// Marshall the results
+		b, err := yaml.Marshal(pod)
+		if err != nil {
+			return err
+		}
+		output = append(output, developmentComment...)
+		output = append(output, b...)
+		// Output the v1.Pod with the v1.Container
+		fmt.Println(string(output))
+	}
+	return nil
+}
+
+// simplePodWithV1Container is a function used by inspect when kube yaml needs to be generated
+// for a single container.  we "insert" that container description in a pod.
+func simplePodWithV1Container(ctr *Container) (*v1.Pod, error) {
```

Comment:
If the container is part of a pod, I think we should check here and bail out with an ErrNotImplemented - we don't do proper pod stuff yet, by the looks of things

Reply:
great suggestion, done